libs/utils/android/workloads/vellamo.py

```diff
@@ -0,0 +1,175 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from android import Workload, Screen, System
+from time import sleep
+
+import os
+import collections
+import copy
+import logging
+
+VELLAMO_DB_PATH = '/data/data/com.quicinc.vellamo/files'
+VELLAMO_SCORE_NAME = 'chapterscores.json'
+VELLAMO_TESTS = ['BROWSER', 'METAL', 'MULTI']
+
+class Vellamo(Workload):
+    """
+    Android Vellamo workload
+    """
+
+    # Package required by this workload
+    package = 'com.quicinc.vellamo'
+    activity = 'com.quicinc.vellamo.main.MainActivity'
+
+    # Setup logger
+    logger = logging.getLogger('Vellamo')
+    logger.setLevel(logging.DEBUG)
+
+    def __init__(self, test_env):
+        super(Vellamo, self).__init__(test_env)
+        self.logger.debug('%14s - Workload created', 'Vellamo')
+
+    def wload_run(self, exp_dir, test_name, collect='', emeter_channel=None):
+        """
+        Run single Vellamo workload. Returns a collection of results.
+
+        :param exp_dir: Path to experiment directory where to store results.
+        :type exp_dir: str
+
+        :param test_name: Name of the test to run
+        :type test_name: str
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
+        :type collect: list(str)
+
+        :param emeter_channel: Energy meter channel from which data is
+            collected
+        :type emeter_channel: str
+        """
+
+        if test_name.upper() not in VELLAMO_TESTS:
+            raise ValueError('Vellamo workload [%s] not supported', test_name)
+
+        # Set parameter depending on test
+        self.logger.debug('Start Vellamo Benchmark [%s]', test_name)
+        test_x, test_y = (0, 0)
+        sleep_time = 0
+        if test_name.upper() == 'BROWSER':
+            test_x, test_y = (970, 740)
```

Comment:
Absolute coords are quite likely not portable across different screen sizes... should we go for percentage values, maybe on some devices this can be just enough ;-)